controllers/util/onboarding_secrets.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ReadPrivateKey(cl client.Client) (*rsa.PrivateKey, error) {
```

Comment:
why can't this function be in existing `provider.go` itself?

Reply:
In Provider.go, their will be import issues for package
"crypto", "crypto/rand", "crypto/rsa". ReadPrivateKey function require "crypto/rsa", "crypto/x509". So for import conflicts, I created an new file.

Reply:
Ack, deleted the onboarding_secrets.go and function called in provider.go